Sources/swift-format/main.swift

```diff
@@ -48,6 +50,41 @@ fileprivate func main(_ arguments: [String]) -> Int32 {
   }
 }
 
+/// Load the configuration.
+private func loadConfiguration(
+  forSwiftFile swiftFilePath: String, configFilePath: String?
+) -> Configuration {
+  if let path = configFilePath {
+    return decodedConfiguration(fromFileAtPath: path)
+  }
+  else {
+    // Search for a ".swift-format" configuration file in the directory of the current .swift file,
+    // or its nearest parent.
+    let swiftFileDir = URL(fileURLWithPath: swiftFilePath).deletingLastPathComponent()
+    return decodedConfiguration(
+      fromFileAtPath: findConfigurationFile(inDirectory: swiftFileDir.path))
+  }
+}
+
+/// Look for a ".swift-format" configuration file in "inDirectory", or its nearest parent. Return
+/// "nil" if one is not found.
+private func findConfigurationFile(inDirectory: String) -> String? {
+  let cwd = FileManager.default.currentDirectoryPath
+  var path = URL(
+    fileURLWithPath: AbsolutePath(inDirectory, relativeTo: AbsolutePath(cwd)).asString)
+  let configFilename = ".swift-format"
+
+  while path.path != "/.." {
```

Comment:
Why is this `"/.."` instead of just `"/"`?

Reply:
This allows me to also check root for a config file. The end of the `while` loop then applies `deletingLastPathComponent()`, turning `"/"` into `"/.."` at which point we exit.

Reply:
That feels a little too subtle. Can you rewrite the loop as a `repeat/while` to move the test to the end of the loop, which should make the check for `"/"` work like you expect?

Reply:
If would have to look like this:
```swift
let testPath = path.appendingPathComponent(configFilename).path
if FileManager.default.fileExists(atPath: testPath) {
  return testPath
}

repeat {
  path = path.deletingLastPathComponent()
  let testPath = path.appendingPathComponent(configFilename).path
  if FileManager.default.fileExists(atPath: testPath) {
    return testPath
  }
} while path.path != "/"
  
```
Unless, I pass the full path to the `.swift` file into the function, instead of its directory.

Reply:
> Unless, I pass the full path to the `.swift` file into the function, instead of its directory.

Yeah, do that, it solves the problem elegantly and I think it's reasonable to have the function's API be "find the `.swift-format` file associated with this `.swift` file" instead of "...starting from this directory."